azure-pipelines-v3.yml

```diff
@@ -72,99 +143,74 @@ jobs:
-        params: https://github.com/MicrosoftDocs/azure-devops-docs-pr --timeout 30 --regression-rules
-      AspNetCore.Docs:
-        params: https://github.com/dotnet/AspNetCore.Docs --profile --timeout 30 --output-html
-      microsoft-365-docs-pr.zh-CN:
-        params: https://github.com/MicrosoftDocs/microsoft-365-docs-pr.zh-CN --branch live-sxs --profile --timeout 25
-      powerbi-docs-pr.de-DE:
-        params: https://github.com/MicrosoftDocs/powerbi-docs-pr.de-DE --branch live-sxs --timeout 25
-      PowerShell-Docs:
-        params: https://github.com/MicrosoftDocs/PowerShell-Docs --timeout 75 --branch staging
-      roslyn-api-docs:
-        params: https://github.com/dotnet/roslyn-api-docs --branch live --profile --timeout 105
-      azure-docs-rest-apis:
-        params: https://github.com/Azure/azure-docs-rest-apis --timeout 130 --error-level Warning
-      dynamics365-docs-odata-apis:
-        params: https://github.com/MicrosoftDocs/dynamics365-docs-odata-apis --timeout 20
-      mc-docs-pr:
-        params: https://github.com/MicrosoftDocs/mc-docs-pr --timeout 70
-      dynamics365smb-devitpro:
-        params: https://github.com/MicrosoftDocs/dynamics365smb-devitpro --timeout 15
-      DevSandbox:
-        params: https://github.com/MicrosoftDocs/DevSandbox --timeout 30
-      test:
-        params: https://github.com/MicrosoftDocs/test --timeout 20 --no-dry-sync
-      DocsRoot:
-        params: https://github.com/MicrosoftDocs/DocsRoot --timeout 20
-      windows-compatibility:
-        params: https://dev.azure.com/cpubwin/windows-compatibility/_git/windows-compatibility --timeout 20
-      azure-docs-cli:
-        params: https://github.com/MicrosoftDocs/azure-docs-cli --timeout 50
-      dataprep-dotnet-pr:
-        params: https://github.com/MicrosoftDocs/dataprep-dotnet-pr --timeout 30
-      azure-mediaplayer-typescript:
-        params: https://github.com/MicrosoftDocs/azure-mediaplayer-typescript --timeout 30
-      quantum-docs-private:
-        params: https://github.com/MicrosoftDocs/quantum-docs-private --timeout 30 --branch main
-      microsoft-graph-docs:
-        params: https://github.com/microsoftgraph/microsoft-graph-docs --timeout 30
-      # dry-runs
-      dryrun.bot-docs-pr:
-        params: https://github.com/MicrosoftDocs/bot-docs-pr --timeout 40 --no-dry-sync --dry-run --public-template
-      dryrun.azure-docs-pr:
-        params: https://github.com/MicrosoftDocs/azure-docs-pr --profile --timeout 50 --dry-run --regression-rules
+      ${{ each repo in parameters.repos }}:
+        ${{ repo.key }}:
+          params: ${{repo.value.params}}
 
   steps:
-  # Install .NET Core sdk
-  - task: UseDotNet@2
-    displayName: 'Install .NET Core sdk $(dotnetVersion)'
-    inputs:
-      packageType: sdk
-      version: $(dotnetVersion)
-
-  # Download RegressionTest binary
-  - download: current
-    displayName: Download RegressionTest binary
-    artifact: RegressionTest
+    - template: prepare.yml
```

Comment:
There doesn't seem to be much value in extracting this `prepare.yml`. The only difference between the two are the _Run Regression Test_ vs _Warmup Regression Test_ steps

Reply:
This template is useful for both the 2 jobs. If we don't extract it, we will have duplicate segments in this pipeline.

Reply:
Template removed.